packages/amplify-graphql-auth-transformer/src/__tests__/resolvers/__snapshots__/subscriptions.test.ts.snap

```diff
@@ -14,18 +14,65 @@ $util.toJson({\\"version\\":\\"2018-05-29\\",\\"payload\\":{}})
 ## [End] Authorization Steps. **"
 `;
 
+exports[`subscriptions generateAuthExpressionForSubscriptions iam renders for iam access 1`] = `
+"## [Start] Authorization Steps. **
+$util.qr($ctx.stash.put(\\"hasAuth\\", true))
+#set( $isAuthorized = false )
+#if( $util.authType() == \\"IAM Authorization\\" )
+  #if( $util.isNull($ctx.identity.cognitoIdentityPoolId) && $util.isNull($ctx.identity.cognitoIdentityId) )
+    $util.qr($ctx.stash.put(\\"hasAuth\\", true))
+    #set( $isAuthorized = true )
+  #else
+$util.unauthorized()
+  #end
+#end
+#if( !$isAuthorized )
+$util.unauthorized()
+#end
+$util.toJson({\\"version\\":\\"2018-05-29\\",\\"payload\\":{}})
+## [End] Authorization Steps. **"
+`;
+
+exports[`subscriptions generateAuthExpressionForSubscriptions iam renders for iam access with roles 1`] = `
+"## [Start] Authorization Steps. **
+$util.qr($ctx.stash.put(\\"hasAuth\\", true))
+#set( $isAuthorized = false )
+#if( $util.authType() == \\"IAM Authorization\\" )
+  #if( $util.isNull($ctx.identity.cognitoIdentityPoolId) && $util.isNull($ctx.identity.cognitoIdentityId) )
+    $util.qr($ctx.stash.put(\\"hasAuth\\", true))
+    #set( $isAuthorized = true )
+  #else
+    #if( !$isAuthorized )
+      #if( $ctx.identity.userArn == $ctx.stash.testClaim )
```

Comment:
Where are we setting this stash variable?

Reply:
here https://github.com/aws-amplify/amplify-category-api/blob/67290951da610989476d366703b7722229eedbf1/packages/amplify-graphql-auth-transformer/src/__tests__/resolvers/subscriptions.test.ts#L29-L35 .

Reply:
addressed in https://github.com/aws-amplify/amplify-category-api/pull/2331/commits/a10d9bbc008fffbfdf691e7e57f33f2e6e7c6403

Reply:
The linked code block doesn't explain where we set the stash variable.

Reply:
The call stack is:
1. https://github.com/aws-amplify/amplify-category-api/blob/11336ca59d035e5b57b62e11c583158291c9b8b4/packages/amplify-graphql-auth-transformer/src/vtl-generator/ddb/resolvers/helpers.ts#L56-L57
2. https://github.com/aws-amplify/amplify-category-api/blob/11336ca59d035e5b57b62e11c583158291c9b8b4/packages/amplify-graphql-auth-transformer/src/vtl-generator/ddb/resolvers/helpers.ts#L165
3. https://github.com/aws-amplify/amplify-category-api/blob/11336ca59d035e5b57b62e11c583158291c9b8b4/packages/amplify-graphql-auth-transformer/src/vtl-generator/ddb/resolvers/subscriptions.ts#L173
4. https://github.com/aws-amplify/amplify-category-api/blob/11336ca59d035e5b57b62e11c583158291c9b8b4/packages/amplify-graphql-auth-transformer/src/__tests__/resolvers/subscriptions.test.ts#L103

![image](https://github.com/aws-amplify/amplify-category-api/assets/5849952/7e978115-9c96-4467-b57b-934dc81a6f7e)

Reply:
Discussed offline.
This appears to be relevant place https://github.com/aws-amplify/amplify-category-api/blob/11336ca59d035e5b57b62e11c583158291c9b8b4/packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts#L1103 .
I'll update test values to match it.

Reply:
Addressed here https://github.com/aws-amplify/amplify-category-api/pull/2331/commits/078fd69f3eac0c8080f144fb3825b2cbc9cfa6cf